cmd/apps/gitea_app.go

```diff
@@ -0,0 +1,145 @@
+// Copyright (c) arkade author(s) 2020. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+package apps
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"path"
+	"strconv"
+	"strings"
+
+	"github.com/alexellis/arkade/pkg/apps"
+	"github.com/alexellis/arkade/pkg/k8s"
+	"github.com/alexellis/arkade/pkg/types"
+
+	"github.com/alexellis/arkade/pkg"
+	"github.com/alexellis/arkade/pkg/config"
+	"github.com/alexellis/arkade/pkg/env"
+	"github.com/alexellis/arkade/pkg/helm"
+	"github.com/sethvargo/go-password/password"
+	"github.com/spf13/cobra"
+)
+
+func MakeInstallGitea() *cobra.Command {
+	var gitea = &cobra.Command{
+		Use:          "gitea",
+		Short:        "Install gitea",
+		Long:         `Install gitea`,
+		Example:      `  arkade install gitea`,
+		SilenceUsage: true,
+	}
+
+	gitea.Flags().Bool("update-repo", true, "Update the helm repo")
+	gitea.Flags().String("namespace", "default", "Kubernetes namespace for the application")
+	gitea.Flags().Bool("persistence", false, "Enable persistence")
+	gitea.Flags().StringP("user", "u", "gitea_admin", "Username of admin user")
+	gitea.Flags().StringP("password", "p", "", "Overide the default random admin-password if this is set")
+	gitea.Flags().StringArray("set", []string{},
+		"Use custom flags or override existing flags \n(example --set persistence.enabled=true)")
+
+	gitea.RunE = func(command *cobra.Command, args []string) error {
+		kubeConfigPath := config.GetDefaultKubeconfig()
+
+		if command.Flags().Changed("kubeconfig") {
+			kubeConfigPath, _ = command.Flags().GetString("kubeconfig")
+		}
+		updateRepo, _ := gitea.Flags().GetBool("update-repo")
+
+		fmt.Printf("Using kubeconfig: %s\n", kubeConfigPath)
+
+		arch := k8s.GetNodeArchitecture()
+		fmt.Printf("Node architecture: %q\n", arch)
+
+		if arch != IntelArch {
```

Comment:
This error will probably be missed.. is there a way to check that against the external DB setting / flag?

Reply:
If anything, you may want to add that at the end.

Reply:
Moved closer to end, and I detect based on flag if a user has set an external database or not (tested on an ARM machine)